test/automated-browser-tests.js

```diff
@@ -0,0 +1,248 @@
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+'use strict';
+
+/* eslint-disable max-len, no-console, padded-blocks, no-multiple-empty-lines */
+/* eslint-env node,mocha */
+
+// These tests make use of selenium-webdriver. You can find the relevant
+// documentation here: http://selenium.googlecode.com/git/docs/api/javascript/index.html
+
+require('chai').should();
+const webdriver = require('selenium-webdriver');
+const chromeOptions = require('selenium-webdriver/chrome');
+const firefoxOptions = require('selenium-webdriver/firefox');
+const which = require('which');
+const del = require('del');
+const testServerHelper = require('./server/server');
+
+
+// Before the tests start, we must build the current files
+GLOBAL.config = {
+  env: 'prod',
+  src: 'test/data/valid-files',
+  dest: 'test/output'
+};
+
+describe('Test WSK in browser', function() {
+  this.timeout(60000);
+
+  // Skip if we are APPVEYOR since we have travis to automate these tests
+  if (process.env.APPVEYOR) {
+    console.log('Skipping WSK browser tests on AppVeyor.');
+    return;
+  }
+
+  // Driver is initialised to null to handle scenarios
+  // where the desired browser isn't installed / fails to load
+  // Null allows afterEach a safe way to skip quiting the driver
+  let globalDriverReference = null;
+  let testServer;
+
+  before(function(done) {
+    // 0 for port will pick a random port number
+    testServerHelper.startServer(0, function(portNumber) {
+      testServer = `http://localhost:${portNumber}`;
+      done();
+    });
+  });
+
+  after(function() {
+    testServerHelper.killServer();
+  });
+
+  afterEach(function(done) {
+    this.timeout(10000);
+
+    // Suggested as fix to 'chrome not reachable'
+    // http://stackoverflow.com/questions/23014220/webdriver-randomly-produces-chrome-not-reachable-on-linux-tests
+    const timeoutGapCb = function() {
+      setTimeout(done, 2000);
+    };
+
+    if (globalDriverReference === null) {
+      return timeoutGapCb();
+    }
+
+    globalDriverReference.quit()
+    .then(() => {
+      globalDriverReference = null;
+      timeoutGapCb();
+    })
+    .thenCatch(() => {
+      globalDriverReference = null;
+      timeoutGapCb();
+    });
+  });
+
+  const performTests = (browserName, driver) => {
+    // The driver methods are wrapped in a new promise because the
+    // selenium-webdriver API seems to using some custom promise
+    // implementation that has slight behaviour differences.
+    return new Promise((resolve, reject) => {
+      driver.get(`${testServer}/test/browser-tests/`)
```

Comment:
selenium driver uses some other Promise library - they aren't native Promises;

// In fact I added a comment just about new Promise on exactly this.....so......